test/should-compile/GHC/TypeLits/Test.hs

```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE DataKinds, RankNTypes, TypeFamilies, TypeOperators #-}
+{-# LANGUAGE DataKinds, RankNTypes, TypeFamilies, TypeOperators, AllowAmbiguousTypes #-}
```

Comment:
Why is this needed?

Reply:
I figured it out: https://github.com/gelisam/typelevel-rewrite-rules/pull/37#discussion_r1138003124

Reply:
Thx!

Reply:
Fixed in #37